### PR TITLE
Document the Function CR

### PR DESCRIPTION
**Pull request by kazydek**
**Description**

Changes proposed in this pull request:

- Added a document describing the Function custom resource (CRD).

**Related issue(s)**
See also #7047

### CODE REVIEW

File: docs/serverless-v2/06-01-function-cr.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+| **spec.function** | Yes | Provides the lambda's source code. |
+
+## Related resources and components
+
+The Function custom resource relies on resources from [Knative Serving](https://knative.dev/docs/serving/) and [Tekton Pipelines](https://github.com/tektoncd/pipeline):
+
+| Resource | Description |
+|----------|-------------|
+|[KService CR](https://github.com/knative/docs/blob/master/docs/serving/spec/knative-api-specification-1.0.md#service) | Orchestrates the deployment and availability of the function.|
+|[TestRun CR](https://github.com/tektoncd/pipeline/blob/master/docs/taskruns.md) | Builds a container image containing the function code together with its configured runtime. |
```
Comment (magicmatatjahu) [inline docs/serverless-v2/06-01-function-cr.md]: Builds a image containing the function code on chosen runtime.

File: docs/serverless-v2/06-01-function-cr.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+These components use this CR:
+
+| Component   |   Description |
+|-------------|---------------|
+| Serverless |  Uses the Function CR for the detailed lambda definition, including the environment on which it should run. |
```
Comment (magicmatatjahu) [inline docs/serverless-v2/06-01-function-cr.md]: image

We don't have `Serverless` component. Can you describe what do you mean?

Comment (magicmatatjahu) [reply docs/serverless-v2/06-01-function-cr.md]: Maybe `Function Controller`?

File: docs/serverless-v2/06-01-function-cr.md [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+| **spec.function** | Yes | Provides the lambda's source code. |
+
+## Related resources and components
+
+The Function custom resource relies on resources from [Knative Serving](https://knative.dev/docs/serving/) and [Tekton Pipelines](https://github.com/tektoncd/pipeline):
+
+| Resource | Description |
+|----------|-------------|
+|[KService CR](https://github.com/knative/docs/blob/master/docs/serving/spec/knative-api-specification-1.0.md#service) | Orchestrates the deployment and availability of the function.|
+|[TestRun CR](https://github.com/tektoncd/pipeline/blob/master/docs/taskruns.md) | Builds a container image with the function code on a chosen runtime. |
```
Comment (magicmatatjahu) [inline docs/serverless-v2/06-01-function-cr.md]: `Builds an image...` :P

File: docs/serverless-v2/06-01-function-cr.md [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+| **spec.function** | Yes | Provides the lambda's source code. |
+
+## Related resources and components
+
+The Function custom resource relies on resources from [Knative Serving](https://knative.dev/docs/serving/) and [Tekton Pipelines](https://github.com/tektoncd/pipeline):
+
+| Resource | Description |
+|----------|-------------|
+|[KService CR](https://github.com/knative/docs/blob/master/docs/serving/spec/knative-api-specification-1.0.md#service) | Orchestrates the deployment and availability of the function.|
+|[TestRun CR](https://github.com/tektoncd/pipeline/blob/master/docs/taskruns.md) | Builds an image with the function code on a chosen runtime. |
```
Comment (bszwarc) [inline docs/serverless-v2/06-01-function-cr.md]:
```suggestion
|[TaskRun CR](https://github.com/tektoncd/pipeline/blob/master/docs/taskruns.md) | Builds an image with the function code on a chosen runtime. |
```

File: docs/serverless-v2/06-01-function-cr.md [hunk 5]
```diff
@@ -0,0 +1,76 @@
+---
+title: Function
+type: Custom Resource
+---
+
+The `functions.serverless.kyma-project.io` CustomResourceDefinition (CRD) is a detailed description of the kind of data and the format used to manage lambdas within Kyma. To get the up-to-date CRD and show the output in the `yaml` format, run this command:
```
Comment (bszwarc) [inline docs/serverless-v2/06-01-function-cr.md]:
```suggestion
The `functions.serverless.kyma-project.io` CustomResourceDefinition (CRD) is a detailed description of the kind of data and the format used to manage lambdas within Kyma. To get the up-to-date CRD and show the output in the YAML format, run this command:
```
AFAIK, we agreed on YAML when talking about format, and `.yaml` when writing about a file extension. However, this part is taken from the template, so 🤔

File: docs/serverless-v2/06-01-function-cr.md [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+| **spec.functionContentType** | No | Specifies the content type of the lambda's code defined in the **function** property. The content type can be either `plaintext` or `base64`. It is set to `plaintext` unless you specify otherwise.|
+| **spec.runtime** | No | Specifies the software runtime used to run the lambda's code. The Function Controller supports `nodejs6` and `nodejs8`. It is set to `nodejs8` unless you specify otherwise. |
+| **spec.timeout** | No | Specifies the duration in seconds after which the lambda execution is terminated. The default value is `180`. |
+| **spec.env** | No | Specifies environment variables you need to export for the lambda. |
+| **spec.deps** | No | Specifies the lambda's dependencies. |
+| **spec.function** | Yes | Provides the lambda's source code. |
+
+## Related resources and components
+
+The Function custom resource relies on resources from [Knative Serving](https://knative.dev/docs/serving/) and [Tekton Pipelines](https://github.com/tektoncd/pipeline):
```
Comment (bszwarc) [inline docs/serverless-v2/06-01-function-cr.md]:
```suggestion
The Function custom resource relies on these resources from [Knative Serving](https://knative.dev/docs/serving/) and [Tekton Pipelines](https://github.com/tektoncd/pipeline):
```

File: docs/serverless-v2/06-01-function-cr.md [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+apiVersion: serverless.kyma-project.io/v1alpha1
+kind: Function
+metadata:
+  name: my-test-lambda
+  namespace: default
+spec:
+  functionContentType: plaintext
+  runtime: nodejs8
+  timeout: 360
+  env:
```
Comment (michal-hudy) [inline docs/serverless-v2/06-01-function-cr.md]: It doesn't work right now ;) Anyway, as you already described it, leave it :)

**Pull request merged**